lib/internal/modules/esm/hooks.js

```diff
@@ -359,10 +344,35 @@ class Hooks {
       }
     }
 
+    if (
+      resolvedImportAssertions != null &&
+      typeof resolvedImportAssertions !== 'object'
+    ) {
+      throw new ERR_INVALID_RETURN_PROPERTY_VALUE(
+        'an object',
+        hookErrIdentifier,
+        'importAssertions',
+        resolvedImportAssertions,
+      );
+    }
+
+    if (
+      format != null &&
+      typeof format !== 'string' // [2]
+    ) {
+      throw new ERR_INVALID_RETURN_PROPERTY_VALUE(
+        'a string',
+        hookErrIdentifier,
+        'format',
+        format,
+      );
+    }
+
     return {
       __proto__: null,
-      format,
       url,
+      importAssertions: resolvedImportAssertions,
+      format,
```

Comment:
What's with the order? I feel like alphabetical makes more sense, no?
```suggestion
      format,
      importAssertions: resolvedImportAssertions,
      url,
```

Reply:
I was putting the required properties first.